### DISCUSSION TITLE
When wrapping a `grid` in a `box` with rounded corners, the `fill` of a grid cell overlaps the border of the `box`

### DISCUSSION THREAD

**Discussion by marc-thieme** [typst/typst#4857] [Q&A]
### Description

![image](https://github.com/user-attachments/assets/fd5e3ef9-d0b6-49d0-9f00-56ed7f05d84a)
The image above is produced by this code:
```typst
#box(radius: 15pt, stroke: red + 5pt, grid(inset: 20pt, fill: red)[text])
```

In my opinion, box and its roundings should confine the contents inside, no matter what they are. That is, the box should
either overwrite the fill of the grid should recognize that it is located in a box with rounded corners and adapt its drawing.

This opinion is based on my use case where I cannot find a way to avoid the corners sticking out and they look very unappealing.

### Operating system

Linux

### Typst version

- [x] I am using the latest version of Typst

**Comment by PgBiel** [comment]
You can use `#box(clip: true, ...)` to achieve this effect.

Although this particular example is a bit contrived as you can add `fill: red` and `inset: 20pt` to the box itself, but I'm not sure if this is exactly what you're trying to do.

**Reply by marc-thieme** [reply]
Thank you very much, this helped. Yes, I reduced the example from my actual usecase in order to make it simplier.

Now, it draws an ever so thin white margin between white line between the box's border and its content:
![image](https://github.com/user-attachments/assets/a928c684-5d18-4b22-93ad-acf362905389)

I can work around this by setting the fill of the box to the same color as the fill of the cell. But in the code below, I have other transparent which are affected by this. I can paint them white in turn, but now I have hardcoded a white background, which isnt optimal either.

```typst
let grit = grid(columns: 2, align: horizon, inset: inset, grid.cell(fill: accent-color, [filled]), grid.vline(stroke: stroke), [transparent])
box(clip: true, stroke: stroke, radius: rounded-corners, grit)
```

**Reply by marc-thieme** [reply]
Is this last issue with the white margin something for which I should open a proper issue?
